### PR TITLE
Rework app / webapp communication

### PR DESCRIPTION
**Pull request by Maxr1998**
Put Kotlin channel inside wrapper class, use channel for communication from NativeInterface, RemotePlayerService and ExternalPlayer.

### CODE REVIEW

File: app/src/main/java/org/jellyfin/mobile/webapp/RemoteVolumeProvider.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import android.media.AudioManager
+import android.media.VolumeProvider
+import org.jellyfin.mobile.utils.Constants
+
+class RemoteVolumeProvider(private val webappFunctionChannel: WebappFunctionChannel) : VolumeProvider(VOLUME_CONTROL_ABSOLUTE, 100, 0) {
+    override fun onAdjustVolume(direction: Int) {
+        when (direction) {
+            AudioManager.ADJUST_RAISE -> {
+                webappFunctionChannel.triggerInputManagerAction(Constants.INPUT_MANAGER_COMMAND_VOL_UP)
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/mobile/webapp/RemoteVolumeProvider.kt:11]: Fix your TODO's

File: app/src/main/java/org/jellyfin/mobile/webapp/RemoteVolumeProvider.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            }
+            AudioManager.ADJUST_LOWER -> {
+                webappFunctionChannel.triggerInputManagerAction(Constants.INPUT_MANAGER_COMMAND_VOL_DOWN)
+                currentVolume -= 2 // TODO: have web notify app with new volume instead
+            }
+        }
+    }
+
+    override fun onSetVolumeTo(volume: Int) {
+        webappFunctionChannel.callPlaybackManager("sendCommand({Name:'SetVolume', Arguments:{Volume:$volume}})")
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/mobile/webapp/RemoteVolumeProvider.kt:22]:
```suggestion
        webappFunctionChannel.callPlaybackManager("sendCommand({ Name: 'SetVolume', Arguments:{ Volume: $volume }})")
```

Would it be worth adding additional methods to the function channel to generate these commands so we don't have hardcoded strings like this one everywhere?

File: app/src/main/java/org/jellyfin/mobile/webapp/RemoteVolumeProvider.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package org.jellyfin.mobile.webapp
+
+import android.media.AudioManager
+import android.media.VolumeProvider
+import org.jellyfin.mobile.utils.Constants
+
+class RemoteVolumeProvider(private val webappFunctionChannel: WebappFunctionChannel) : VolumeProvider(VOLUME_CONTROL_ABSOLUTE, 100, 0) {
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/mobile/webapp/RemoteVolumeProvider.kt:7]:
```suggestion
class RemoteVolumeProvider(
    private val webappFunctionChannel: WebappFunctionChannel
) : VolumeProvider(VOLUME_CONTROL_ABSOLUTE, 100, 0) {
```

I prefer multi-line constructors if they become too long

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/jellyfin/mobile/webapp/RemoteVolumeProvider.kt (hunk 1 above)
Comment (Maxr1998) [reply app/src/main/java/org/jellyfin/mobile/webapp/RemoteVolumeProvider.kt:11]: They were already there in master xD

File: app/src/main/java/org/jellyfin/mobile/webapp/RemoteVolumeProvider.kt (hunk 2 above)
Comment (Maxr1998) [reply app/src/main/java/org/jellyfin/mobile/webapp/RemoteVolumeProvider.kt:22]: Yeah, I think that makes sense..

File: app/src/main/java/org/jellyfin/mobile/webapp/RemoteVolumeProvider.kt (hunk 1 above)
Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/mobile/webapp/RemoteVolumeProvider.kt]: Is it something we can fix or is it not possible without changing webui?

**Review by nielsvanvelzen — APPROVED** [review]

File: app/src/main/java/org/jellyfin/mobile/webapp/RemoteVolumeProvider.kt (hunk 1 above)
Comment (Maxr1998) [reply app/src/main/java/org/jellyfin/mobile/webapp/RemoteVolumeProvider.kt]: It needs changes to the web UI, unfortunately. But then again, we'll probably have a native UI before we get those changes, lol.

**Pull request merged**